src/components/Export.tsx

```diff
@@ -0,0 +1,358 @@
+        break;
+      case ExportMode.CURRENT:
+        min = props.currentFrame;
+        max = props.currentFrame;
+        break;
+      case ExportMode.CUSTOM:
+        // Clamp range values in case of unsafe input
+        min = clamp(customMin, 0, props.totalFrames - 1);
+        max = clamp(customMax, min, props.totalFrames - 1);
+    }
+
+    // Start the recording
+    props.startRecording({
+      min: min,
+      max: max,
+      prefix: imagePrefix,
+      minDigits: (props.totalFrames - 1).toString().length,
+      frameIncrement: frameIncrement,
+      onCompletedCallback: () => {
+        // Close modal once recording finishes and show completion notification
+        setPercentComplete(100);
+        notification.success({
+          message: "Export complete.",
+          placement: "bottomLeft",
+          duration: 4,
+          icon: <CheckCircleOutlined style={{ color: theme.color.text.success }} />,
+        });
+        // Close the modal after a small delay so the success notification can be seen
+        setIsPlayingCloseAnimation(true);
+        setTimeout(() => stopRecording(true), 750);
+      },
+      onRecordedFrameCallback: (frame: number) => {
+        // Update the progress bar as frames are recorded.
+        setPercentComplete(Math.floor(((frame - min) / (max - min)) * 100));
+      },
+    });
+  };
+
+  const numExportedFrames = Math.max(Math.ceil((customMax - customMin + 1) / frameIncrement), 1);
+
+  return (
+    <div ref={modalContextRef}>
+      {/* Export button */}
+      <Button
+        type="primary"
+        onClick={() => setIsLoadModalOpen(true)}
+        disabled={props.disabled}
+        data-testid={TEST_ID_OPEN_EXPORT_MODAL_BUTTON}
+      >
+        Export
+      </Button>
+
+      {/* Export modal */}
+      <Modal
+        title={"Export image sequence"}
+        open={isLoadModalOpen}
+        onCancel={handleCancel}
+        cancelButtonProps={{ hidden: true }}
+        centered={true}
+        // Don't allow cancellation of modal by clicking off it when the recording is happening
+        maskClosable={!isRecording}
+        getContainer={modalContextRef.current || undefined}
+        footer={
+          // Layout: Optional Progress meter - Export/Stop Button - Cancel Button
+          <HorizontalDiv style={{ flexDirection: "row", alignItems: "center", justifyContent: "flex-end" }}>
+            {(percentComplete !== 0 || isRecording) && (
+              <Tooltip title={percentComplete + "%"} style={{ verticalAlign: "middle" }}>
+                <Progress
+                  style={{ marginRight: "8px", verticalAlign: "middle" }}
+                  type="circle"
+                  size={theme.controls.heightSmall - 6}
+                  percent={percentComplete}
+                  showInfo={false}
+                  strokeColor={percentComplete === 100 ? theme.color.text.success : theme.color.theme}
+                  strokeWidth={12}
+                />
+              </Tooltip>
```

Comment:
Kind of wild of Ant to make you put the code for what goes at the bottom of the modal before the code for the content, but what can you do

Reply:
Ant moment lol 🐜